cli/src/main/java/org/jboss/as/cli/handlers/DeploymentInfoHandler.java

```diff
@@ -405,4 +406,14 @@ protected void handleResponse(CommandContext ctx, ModelNode response, boolean co
             otherServerGroups = null;
         }
     }
+
+    @Override
+    protected void addHeaders(CommandContext ctx, ModelNode request) throws CommandFormatException {
+        //WFLY-7001
+        super.addHeaders(ctx, request);
+        final ModelNode opHeaders = request.get(Util.OPERATION_HEADERS);
```

Comment:
Hi @baranowb, thank you for taking over on this one. It is nice, one remark. 
ModelNode.get is mutating the request. So although there is no headers, after get is called, an headers (of type OBJECT) will be added.
You could call isDefined(Util.OPERATION_HEADERS).
if (request.isDefined(Util.OPERATION_HEADERS) && request.get(Util.OPERATION_HEADERS).getType() == ModelType.BOOLEAN) { ...